### PR TITLE
Slight refactoring of cross-reference table entries

### PR DESCRIPTION
**Pull request by noavarice**
Also, added license header in test file. If it has appropriate format, I will add it to the files where I missed to add it

### CODE REVIEW

File: openpdf/pom.xml [hunk 1]
```diff
@@ -51,6 +51,11 @@
             <version>${jupiter.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
```
Comment (andreasrosdal) [inline openpdf/pom.xml]: Why add another dependency? Perhaps test scope only?

Comment (noavarice) [reply openpdf/pom.xml]: Yes, forgot it while making changes

Comment (noavarice) [reply openpdf/pom.xml]: Fixed

**Comment by andreasrosdal** [comment]
Why are you refactoring the cross-reference table entries ?

**Comment by andreasrosdal** [comment]
Does this change break the API for existing users of OpenPDF?

**Comment by noavarice** [comment]
> Why are you refactoring the cross-reference table entries ?

To clarify and document existing behaviour as well as add tests for it

**Comment by noavarice** [comment]
> Does this change break the API for existing users of OpenPDF?

No, it should not

**Comment by andreasrosdal** [comment]
Thanks. 

Please help with #128.

**Pull request merged**